docs/csharp/roslyn-sdk/tutorials/how-to-write-csharp-analyzer-code-fix.md

```diff
@@ -243,114 +252,53 @@ The final step is to make your edit. There are three steps to this process:
-
-```csharp
-[DataTestMethod]
-[DataRow("")]
-public void WhenTestCodeIsValidNoDiagnosticIsTriggered(string testCode)
-{
-    VerifyCSharpDiagnostic(testCode);
-}
-```
-
-You can create a new data row for this test by defining any code fragment that should not cause your diagnostic to trigger a warning. This overload of `VerifyCSharpDiagnostic` passes when there are no diagnostics triggered for the source code fragment.
-
-Next, replace `TestMethod2` with this test that ensures a diagnostic is raised and a code fix applied for the source code fragment:
+Open the *MakeConstUnitTests.cs* file in the unit test project. The template created two tests that follow the two common patterns for an analyzer and code fix unit test. `TestMethod1` shows the pattern for a test that ensures the analyzer doesn't report a diagnostic when it shouldn't. `TestMethod2` shows the pattern for reporting a diagnostic and running the code fix.
 
-```csharp
-[DataTestMethod]
-[DataRow(LocalIntCouldBeConstant, LocalIntCouldBeConstantFixed, 10, 13)]
-public void WhenDiagnosticIsRaisedFixUpdatesCode(
-    string test,
-    string fixTest,
-    int line,
-    int column)
-{
-    var expected = new DiagnosticResult
-    {
-        Id = MakeConstAnalyzer.DiagnosticId,
-        Message = new LocalizableResourceString(nameof(MakeConst.Resources.AnalyzerMessageFormat), MakeConst.Resources.ResourceManager, typeof(MakeConst.Resources)).ToString(),
-        Severity = DiagnosticSeverity.Warning,
-        Locations =
-            new[] {
-                    new DiagnosticResultLocation("Test0.cs", line, column)
-                }
-    };
-
-    VerifyCSharpDiagnostic(test, expected);
-
-    VerifyCSharpFix(test, fixTest);
-}
-```
+The template uses [Microsoft.CodeAnalysis.Testing](https://github.com/dotnet/roslyn-sdk/blob/master/src/Microsoft.CodeAnalysis.Testing/README.md) packages for unit testing.
 
-The preceding code also made a couple changes to the code that builds the expected diagnostic result. It uses the public constants registered in the `MakeConst` analyzer. In addition, it uses two string constants for the input and fixed source. Add the following string constants to the `UnitTest` class:
+Add the following test method to the `MakeConstUnitTest` class:
 
-[!code-csharp[string constants for fix test](~/samples/snippets/csharp/roslyn-sdk/Tutorials/MakeConst/MakeConst.Test/MakeConstUnitTests.cs#FirstFixTest "string constants for fix test")]
+[!code-csharp[test method for fix test](snippets/how-to-write-csharp-analyzer-code-fix/MakeConst/MakeConst.Test/MakeConstUnitTests.cs#FirstFixTest "test method for fix test")]
 
-Run these two tests to make sure they pass. In Visual Studio, open the **Test Explorer** by selecting **Test** > **Windows** > **Test Explorer**. Then select the **Run All** link.
+Run these two tests to make sure they pass. In Visual Studio, open the **Test Explorer** by selecting **Test** > **Windows** > **Test Explorer**. Then select the **Run All** option.
```

Comment:
```suggestion
Run these two tests to make sure they pass. In Visual Studio, open the **Test Explorer** by selecting **Test** &rarr; **Windows** &rarr; **Test Explorer**. Then select the **Run All** option.
```

Reply:
Special characters like this are "mostly" not encoded across the repository. Is there to prefer encoding over the actual character?

Reply:
I've never used the encoded versions.

Reply:
I would've used `&gt;` (greater-than), but if rarr looks the same, that's fine.

Reply:
Ah I just noticed it's different. `&arr;` is `→`. But I'm not sure if there are guidelines for docs to prefer `>`?

Reply:
Yes, the guideline for successive menus is to use `>`. It's also preferred not to refer to the type of the control, so the last sentence should be `Then select **Run All**.`.

Reply:
```suggestion
Run these two tests to make sure they pass. In Visual Studio, open the **Test Explorer** by selecting **Test** > **Windows** > **Test Explorer**. Then select **Run All**.
```